common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -465,7 +470,11 @@ public static void copyMemory(long srcAddr, byte[] dst, int dstIndex, long lengt
      * consumer (one thread!).
      */
     public static <T> Queue<T> newMpscQueue() {
-        return new MpscLinkedQueue<T>();
+        if (hasUnsafe()) {
+            return JAVA_VERSION < 8 ? new MpscLinkedQueue7<T>() : new MpscLinkedQueue8<T>();
```

Comment:
Actually we beed to support java5 as well. That said I think it is fine to just handle it the same way as if unsafe is not present.

@nitsanw I think you told me we should use a different queue. Can you jump in?